pkg/oplabels/labels.go

```diff
@@ -43,3 +45,12 @@ func HasManagedByCassandraOperatorLabel(m map[string]string) bool {
 	v, ok := m[ManagedByLabel]
 	return ok && v == ManagedByLabelValue
 }
+
+// CleanLabelName a valid label must be an empty string or consist of alphanumeric characters,
+// '-', '_' or '.', and must start and end with an alphanumeric.
+// Note: we apply a prefix of "cassandra-" to the cluster name value used as label name.
+// As such, empty string isn't a valid case.
+func CleanLabelName(labelName string) string {
+	trimLeft := strings.TrimLeft(blacklistRegex.ReplaceAllString(labelName, emptyString), labelCut)
```

Comment:
We shouldn't do trimming as the regexp itself can do that. Also, instead of removing blacklisted characters, the regexp should take care of adding all accepted characters in it. So, find all accepted characters and then join them together.

So, regexp.FindAll(regexp) and then strings.Join(regexpResult)

Reply:
It doesn't matter if they're easier for developer to read or not, whatever anyone has currently deployed in any cluster must still work.

If I have `My_-.Cluster`, then that must work. I fail to see why we would trim if regexp trims them also (if they're not accepted characters, they're not copied). What makes it easier to understand that we do something in two places?

Reply:
I'll flip the regex to use of whitelist pattern instead of a blacklist in the next push.  

Using this approach, I believe it still requires removal of the spaces before applying the regex (or build into the regex pattern itself), as I noticed that in the following case, the accepted chars were ignored when spaces were present.  

Usage: 
(([A-Za-z0-9][-A-Za-z0-9_.]*)?[A-Za-z0-9])?
```golang
var unclean = "+!*(-_)cor @#$%^&rect _ LABEL.name-1=<>_?,."
```

Results in:
`correctLABEL.name-1`, but expecting `correct_LABEL.name-1`